octopusdeploy_framework/resource_channel.go

```diff
@@ -0,0 +1,275 @@
+	if err != nil {
+		resp.Diagnostics.AddError("Error deleting channel", err.Error())
+		return
+	}
+}
+
+func (*channelResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func expandChannel(ctx context.Context, model schemas.ChannelModel) *channels.Channel {
+	var channelName = model.Name.ValueString()
+	var projectId = model.ProjectId.ValueString()
+
+	channel := channels.NewChannel(channelName, projectId)
+
+	channel.ID = model.ID.ValueString()
+	channel.Description = model.Description.ValueString()
+	channel.IsDefault = model.IsDefault.ValueBool()
+	channel.LifecycleID = model.LifecycleId.ValueString()
+	channel.Rules = expandChannelRules(model.Rule)
+	channel.SpaceID = model.SpaceId.ValueString()
+	channel.TenantTags = expandStringList(model.TenantTags)
+
+	return channel
+}
+
+func expandChannelRules(rules types.List) []channels.ChannelRule {
+	if rules.IsNull() || rules.IsUnknown() {
+		return nil
+	}
+
+	var rulesMap []map[string]interface{}
+	rules.ElementsAs(context.Background(), &rulesMap, false)
+
+	var channelRules []channels.ChannelRule
+	for _, ruleMap := range rulesMap {
+		channelRule := expandChannelRule(ruleMap)
+		channelRules = append(channelRules, channelRule)
+	}
+
+	return channelRules
+}
+
+func expandChannelRule(rule map[string]interface{}) channels.ChannelRule {
+	var channelRule channels.ChannelRule
+
+	channelRule.ID = rule["id"].(string)
+	actionPackage := rule["action_package"].([]map[string]interface{})
+	channelRule.ActionPackages = expandChannelRuleDeploymentActionPackages(actionPackage)
+	channelRule.Tag = rule["tag"].(string)
+	channelRule.VersionRange = rule["version_range"].(string)
+
+	return channelRule
+}
+
+func expandChannelRuleDeploymentActionPackages(actionPackages []map[string]interface{}) []packages.DeploymentActionPackage {
+	var actionPackagesExpanded []packages.DeploymentActionPackage
+	for _, actionPackage := range actionPackages {
+		actionPackageExpanded := expandChannelRuleDeploymentActionPackage(actionPackage)
+		actionPackagesExpanded = append(actionPackagesExpanded, actionPackageExpanded)
+	}
+	return actionPackagesExpanded
+}
+
+func expandChannelRuleDeploymentActionPackage(actionPackageMap map[string]interface{}) packages.DeploymentActionPackage {
+	return packages.DeploymentActionPackage{
+		DeploymentAction: actionPackageMap["deployment_action"].(string),
+		PackageReference: actionPackageMap["package_reference"].(string),
+	}
+
+}
+
+func flattenChannel(ctx context.Context, channel *channels.Channel, model schemas.ChannelModel) schemas.ChannelModel {
+	model.ID = types.StringValue(channel.GetID())
+	model.Description = types.StringValue(channel.Description)
+
+	if !channel.IsDefault && model.IsDefault.IsNull() {
+		model.IsDefault = types.BoolNull()
+	} else {
+		model.IsDefault = types.BoolValue(channel.IsDefault)
+	}
```

Comment:
From memory and by looking at other resources we've migrated to plugin framework, I think you could change this to simply:
```suggestion
        model.IsDefault = types.BoolValue(channel.IsDefault)
```